src/TheMovie/TheMovie/Services/MovieService.cs

```diff
@@ -0,0 +1,30 @@
+﻿using Newtonsoft.Json;
+using System.Threading.Tasks;
+using TheMovie.Common.API;
+using TheMovie.Models;
+
+namespace TheMovie.Services
+{
+    public class MovieService : IMovieService
+    {
+        private readonly IHttpClientService _httpClientService;
+
+        public MovieService(IHttpClientService httpClientService)
+        {
+            _httpClientService = httpClientService;
+        }
+
+        public async Task<UpcomingMoviesModel> GetAll(int page, string language, string region)
+        {
+            var response = await _httpClientService.HeaderTheMovieApi().GetAsync($"{TheMovieApi.API_URL}movie/upcoming?api_key={TheMovieApi.API_KEY}&page={page}&language={language}&region={region}");
+
+            if (response.IsSuccessStatusCode)
+            {
+                var stringResponse = await response.Content.ReadAsStringAsync();
+                return JsonConvert.DeserializeObject<UpcomingMoviesModel>(stringResponse);
+            }
+
+            return null;
```

Comment:
The same comment here regarding the null check.

Reply:
I did really forget to check if the data was empty :(